src/index.tsx

```diff
@@ -1,10 +1,37 @@
 import ReactDOM from 'react-dom';
-import { HashRouter } from 'react-router-dom';
+import {
+  HashRouter,
+  Navigate,
+  Route,
+  Routes,
+} from 'react-router-dom';
 import { App } from './App';
+import { TabsPage } from './pages/TabsPage';
+import { HomePage } from './pages/HomePage';
+import { NotFoundPage } from './pages/NotFoundPage';
 
 ReactDOM.render(
   <HashRouter>
-    <App />
+    <Routes>
+      <Route path="/" element={<App />}>
+        <Route>
+          <Route path="/" element={<HomePage />} />
+          <Route path="/home" element={<Navigate to="/" />} />
+          <Route
+            path="/tabs"
+          >
+            <Route
+              index
+              element={<TabsPage />}
+            />
+
+            <Route path="/tabs:tabId" element={<TabsPage />} />
+
+          </Route>
+          <Route path="*" element={<NotFoundPage />} />
+        </Route>
+      </Route>
+    </Routes>
```

Comment:
It's not the correct way to structure your app.
1. Use a router inside your APP component. And use the app as the root of your application.
2. For layout create a `Layout` component that will render the view part of your application
3. This layout component should render a `Navigation` and wrap the content to a section. You can use Outlet from react-router-dom.

```
  <Navigation />
  <main className="section">
    <Outlet />
  </main>
```

```suggestion
<Routes>
  <Route path="/" element={<Layout />}>
    .. homepage
    ...tabs
    .. etc
</Routes>     
```